File: src/main/node/net/ecma/Library/Examples/ICTS/stage/intro.mjs
'use strict';
var X3D = require('./x3d.mjs');
try {
	var fs = require('fs');
} catch (e) {
console.log("Problems loading fs. On browser?",e);
}
var SFString = require('./x3d.mjs');
var SFNode = require('./x3d.mjs');
var head = require('./x3d.mjs');
var MFNode = require('./x3d.mjs');
var meta = require('./x3d.mjs');
var Scene = require('./x3d.mjs');
var WorldInfo = require('./x3d.mjs');
var MFString = require('./x3d.mjs');
var Transform = require('./x3d.mjs');
var Collision = require('./x3d.mjs');
var SFBool = require('./x3d.mjs');
var SFVec3f = require('./x3d.mjs');
var SFRotation = require('./x3d.mjs');
var Shape = require('./x3d.mjs');
var Appearance = require('./x3d.mjs');
var Material = require('./x3d.mjs');
var SFFloat = require('./x3d.mjs');
var SFColor = require('./x3d.mjs');
var IndexedLineSet = require('./x3d.mjs');
var MFInt32 = require('./x3d.mjs');
var Coordinate = require('./x3d.mjs');
var MFVec3f = require('./x3d.mjs');
var X3D0 =  new X3D({

      profile : new SFString("Full"),
      version : new SFString("3.3"),
      head : new SFNode(
        new head({
          meta : new MFNode([
            new meta({
              name : new SFString("comment"),
              content : new SFString("World of Titania")}),

            new meta({
              name : new SFString("created"),
              content : new SFString("Thu, 23 Apr 2015 06:07:09 GMT")}),

            new meta({
              name : new SFString("creator"),
              content : new SFString("Holger Seelig")}),

            new meta({
              name : new SFString("generator"),
              content : new SFString("Titania V0.7.8, http://titania.create3000.de")}),

            new meta({
              name : new SFString("modified"),
              content : new SFString("Thu, 23 Apr 2015 06:07:09 GMT")})])})),
      Scene : new SFNode(
        new Scene({
          children : new MFNode([
            new WorldInfo({
              info : new MFString([", ","Created in TitaniaPackaged by CosmoPackage"])}),

            new Transform({
              DEF : new SFString("nc"),
              children : new MFNode([
                new Collision({
                  enabled : new SFBool(false),
                  proxy : new SFNode(
                    new Transform({
                      children : new MFNode([
                        new Transform({
                          translation : new SFVec3f([-1.37383,0.503751,0]),
                          rotation : new SFRotation([0,0,-1,1.5708]),
                          children : new MFNode([
                            new Transform({
                              DEF : new SFString("_1"),
                              children : new MFNode([
                                new Transform({
                                  translation : new SFVec3f([0,0,6.63116]),
                                  children : new MFNode([
                                    new Transform({
                                      DEF : new SFString("_2"),
                                      scale : new SFVec3f([1,2.98486,1]),
                                      children : new MFNode([
                                        new Shape({
                                          appearance : new SFNode(
                                            new Appearance({
                                              material : new SFNode(
                                                new Material({
                                                  ambientIntensity : new SFFloat(0),
                                                  diffuseColor : new SFColor([0,0,0]),
                                                  emissiveColor : new SFColor([0.819149,0.686645,0.392914]),
                                                  shininess : new SFFloat(0)}))})),
                                          geometry : new SFNode(
                                            new IndexedLineSet({
                                              coordIndex : new MFInt32([0,1,3,2,-1,4,5,7,6,4,-1,6,7,1,0,-1,2,3,5,4,-1,6,0,2,4,-1,1,7,5,3,-1]),
                                              coord : new SFNode(
                                                new Coordinate({
                                                  point : new MFVec3f([-0.5,0.5,0.5,-0.5,-0.5,0.5,0.5,0.5,0.5,0.5,-0.5,0.5,0.5,0.5,-0.5,0.5,-0.5,-0.5,-0.5,0.5,-0.5,-0.5,-0.5,-0.5])}))}))})])})])}),

                                new Transform({
                                  translation : new SFVec3f([0,0,2.57238]),
                                  children : new MFNode([
                                    new Transform({
                                      USE : new SFString("_2")})])}),

                                new Transform({
                                  translation : new SFVec3f([0,0,-6.72781]),
                                  children : new MFNode([
                                    new Transform({
                                      USE : new SFString("_2")})])}),

                                new Transform({
                                  translation : new SFVec3f([-0.807327,1.49278,4.45628]),
                                  rotation : new SFRotation([0.0000398219,1,-0.0000398219,4.71239]),
                                  scaleOrientation : new SFRotation([0.0000361377,1,-0.0000237345,2.35619]),
                                  children : new MFNode([
                                    new Transform({
                                      USE : new SFString("_2")})])}),

                                new Transform({
                                  translation : new SFVec3f([-0.715973,0.800763,-2.54796]),
                                  rotation : new SFRotation([0.0000398219,1,-0.0000398219,4.71239]),
                                  scaleOrientation : new SFRotation([0.0000361377,1,-0.0000237345,2.35619]),
                                  children : new MFNode([
                                    new Transform({
                                      USE : new SFString("_2")})])}),

                                new Transform({
                                  USE : new SFString("_2")})])})])}),

                        new Transform({
                          USE : new SFString("_1")})])}))})])})])}))});
console.log(X3D0.toXMLNode());

File: src/main/cplusplus/net/coderextreme/Library/Examples/ICTS/stage/intro.cpp
#define False false
#define True true
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/IndexedFaceSetExample/pch.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/IndexedFaceSetExample/framework.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/include/glut.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/include/X3DLib.h"
int main(int argc, char ** argv) {
X3D& X3D0 =  X3D();
X3D0.setProfile(CString("Full"));
X3D0.setVersion(CString("3.3"));
head& head1 =  head();
meta& meta2 =  meta();
meta2.setName(CString("comment"));
meta2.setContent(CString("World of Titania"));
head1.addMeta(&meta2);

meta& meta3 =  meta();
meta3.setName(CString("created"));
meta3.setContent(CString("Thu, 23 Apr 2015 06:07:09 GMT"));
head1.addMeta(&meta3);

meta& meta4 =  meta();
meta4.setName(CString("creator"));
meta4.setContent(CString("Holger Seelig"));
head1.addMeta(&meta4);

meta& meta5 =  meta();
meta5.setName(CString("generator"));
meta5.setContent(CString("Titania V0.7.8, http://titania.create3000.de"));
head1.addMeta(&meta5);

meta& meta6 =  meta();
meta6.setName(CString("modified"));
meta6.setContent(CString("Thu, 23 Apr 2015 06:07:09 GMT"));
head1.addMeta(&meta6);

X3D0.setHead(&head1);

Scene& Scene7 =  Scene();
WorldInfo& WorldInfo8 =  WorldInfo();
WorldInfo8.setInfo(new CString[]{CString(", "), CString("Created in TitaniaPackaged by CosmoPackage")}, 2);
Scene7.addChild(&WorldInfo8);

Transform& Transform9 =  Transform();
Transform9.setDEF(CString("nc"));
Collision& Collision10 =  Collision();
Collision10.setEnabled(False);
Transform& Transform11 =  Transform();
Transform& Transform12 =  Transform();
Transform12.setTranslation(new float[]{-1.37383,0.503751,0});
Transform12.setRotation(new float[]{0,0,-1,1.5708});
Transform& Transform13 =  Transform();
Transform13.setDEF(CString("_1"));
Transform& Transform14 =  Transform();
Transform14.setTranslation(new float[]{0,0,6.63116});
Transform& Transform15 =  Transform();
Transform15.setDEF(CString("_2"));
Transform15.setScale(new float[]{1,2.98486,1});
Shape& Shape16 =  Shape();
Appearance& Appearance17 =  Appearance();
Material& Material18 =  Material();
Material18.setAmbientIntensity(0);
Material18.setDiffuseColor(new float[]{0,0,0});
Material18.setEmissiveColor(new float[]{0.819149,0.686645,0.392914});
Material18.setShininess(0);
Appearance17.addChild(&Material18);

Shape16.addChild(&Appearance17);

IndexedLineSet& IndexedLineSet19 =  IndexedLineSet();
IndexedLineSet19.setCoordIndex(new int32_t[]{0,1,3,2,-1,4,5,7,6,4,-1,6,7,1,0,-1,2,3,5,4,-1,6,0,2,4,-1,1,7,5,3,-1}, 31);
Coordinate& Coordinate20 =  Coordinate();
Coordinate20.setPoint(new float[]{-0.5,0.5,0.5,-0.5,-0.5,0.5,0.5,0.5,0.5,0.5,-0.5,0.5,0.5,0.5,-0.5,0.5,-0.5,-0.5,-0.5,0.5,-0.5,-0.5,-0.5,-0.5}, 24);
IndexedLineSet19.setCoord(&Coordinate20);

Shape16.setGeometry(&IndexedLineSet19);

Transform15.addChild(&Shape16);

Transform14.addChild(&Transform15);

Transform13.addChild(&Transform14);

Transform& Transform21 =  Transform();
Transform21.setTranslation(new float[]{0,0,2.57238});
Transform& Transform22 =  Transform();
Transform22.setUSE(CString("_2"));
Transform21.addChild(&Transform22);

Transform13.addChild(&Transform21);

Transform& Transform23 =  Transform();
Transform23.setTranslation(new float[]{0,0,-6.72781});
Transform& Transform24 =  Transform();
Transform24.setUSE(CString("_2"));
Transform23.addChild(&Transform24);

Transform13.addChild(&Transform23);

Transform& Transform25 =  Transform();
Transform25.setTranslation(new float[]{-0.807327,1.49278,4.45628});
Transform25.setRotation(new float[]{0.0000398219,1,-0.0000398219,4.71239});
Transform25.setScaleOrientation(new float[]{0.0000361377,1,-0.0000237345,2.35619});
Transform& Transform26 =  Transform();
Transform26.setUSE(CString("_2"));
Transform25.addChild(&Transform26);

Transform13.addChild(&Transform25);

Transform& Transform27 =  Transform();
Transform27.setTranslation(new float[]{-0.715973,0.800763,-2.54796});
Transform27.setRotation(new float[]{0.0000398219,1,-0.0000398219,4.71239});
Transform27.setScaleOrientation(new float[]{0.0000361377,1,-0.0000237345,2.35619});
Transform& Transform28 =  Transform();
Transform28.setUSE(CString("_2"));
Transform27.addChild(&Transform28);

Transform13.addChild(&Transform27);

Transform& Transform29 =  Transform();
Transform29.setUSE(CString("_2"));
Transform13.addChild(&Transform29);

Transform12.addChild(&Transform13);

Transform11.addChild(&Transform12);

Transform& Transform30 =  Transform();
Transform30.setUSE(CString("_1"));
Transform11.addChild(&Transform30);

Collision10.addChildren(&Transform11);

Transform9.addChild(&Collision10);

Scene7.addChild(&Transform9);

X3D0.setScene(&Scene7);

}
